File: app/src/main/java/org/worshipsongs/preference/FontDialogPreference.java
package org.worshipsongs.preference;

import android.content.Context;
import android.content.SharedPreferences;
import android.content.res.TypedArray;
import android.preference.Preference;
import android.preference.PreferenceManager;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.SeekBar;
import android.widget.TextView;

import org.worshipsongs.worship.R;

/**
 * Author:Madasamy
 * version:1.0.0
 */
public class FontDialogPreference extends Preference
{
    private int fontSize;
    private int defaultFontSize;
    private int maxSize = 100;
    private SharedPreferences customSharedPreference = PreferenceManager.getDefaultSharedPreferences(FontDialogPreference.this.getContext());
    private TextView fontSizetextView;

    public FontDialogPreference(Context context, AttributeSet attrs)
    {
        super(context, attrs);
        setPersistent(true);
        if (attrs != null) {
            maxSize = attrs.getAttributeIntValue(null, "maxSize", 20);
        }
    }

    @Override
    protected View onCreateView(ViewGroup parent)
    {
        LayoutInflater layoutInflater = (LayoutInflater) getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View view = layoutInflater.inflate(R.layout.font_size_dialog, parent, false);
        setFontSizeSeekBar(view);
        setFontSizeTextView(view);
        setMaxSizeTextView(view);
        return view;
    }

    private void setFontSizeSeekBar(View view)
    {
        SeekBar fontSizeSeekBar = (SeekBar) view.findViewById(R.id.portrait_font_size);
        fontSizeSeekBar.setMax(maxSize);
        fontSize = customSharedPreference.getInt(getKey(), defaultFontSize);
        fontSizeSeekBar.setProgress(fontSize);
        fontSizeSeekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener()
        {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser)
            {
                fontSize = progress;
                String text = getContext().getResources().getString(R.string.fontSize) + ": " + fontSize;
                fontSizetextView.setText(text);
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar)
            {

            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar)
            {
                saveFontSizePreference(getKey(), fontSize);
            }
        });
    }

    private void setFontSizeTextView(View view)
    {
        fontSizetextView = (TextView) view.findViewById(R.id.fontSize);
        String text = getContext().getResources().getString(R.string.fontSize) + ": " + fontSize;
        fontSizetextView.setText(text);
    }

    private void setMaxSizeTextView(View view)
    {
        TextView textView = (TextView) view.findViewById(R.id.maxsize_textView);
        textView.setText(String.valueOf(maxSize));
    }

    @Override
    protected void onSetInitialValue(boolean restore, Object defaultValue)
    {
        setDefaultValue(restore ? getPersistedInt(20) : (Integer) defaultValue);
    }

    @Override
    protected Object onGetDefaultValue(TypedArray a, int index)
    {
        defaultFontSize = a.getInteger(index, 20);
        return defaultFontSize;
    }

    private void saveFontSizePreference(String key, int fontSize)
    {
        SharedPreferences fontSizePreference = PreferenceManager.getDefaultSharedPreferences(FontDialogPreference.this.getContext());
        SharedPreferences.Editor editor = fontSizePreference.edit();
        editor.putInt(key, fontSize);
        editor.apply();
    }
}

File: app/src/main/java/org/worshipsongs/preference/FontDialogPreference.kt
package org.worshipsongs.preference

import android.content.Context
import android.util.AttributeSet
import android.widget.SeekBar
import android.widget.TextView
import androidx.preference.Preference
import androidx.preference.PreferenceManager
import androidx.preference.PreferenceViewHolder
import kotlinx.android.synthetic.main.font_size_dialog.view.*
import org.worshipsongs.R

/**
 * @author Madasamy
 * @version 1.0.0
 */
class FontDialogPreference(context: Context, attrs: AttributeSet?) : Preference(context, attrs)
{
    private var fontSize: Int = 0
    private var defaultFontSize = 20
    private var maxSize = 100
    private val customSharedPreference = PreferenceManager.getDefaultSharedPreferences(this@FontDialogPreference.context)

    init
    {
        layoutResource = R.layout.font_size_dialog
        isPersistent = true
        if (attrs != null)
        {
            maxSize = attrs.getAttributeIntValue(null, "maxSize", 100)
            defaultFontSize = attrs.getAttributeIntValue(null, "fontSize", 20)
        }
    }

    override fun onBindViewHolder(holder: PreferenceViewHolder?)
    {
        super.onBindViewHolder(holder)
        with(holder!!.itemView){
           setFontSizeSeekBar(portrait_font_size, fontSize)
            setFontSizeTextView(fontSize)
            maxsize_textView.text = maxSize.toString()
        }
    }


    private fun setFontSizeSeekBar(fontSizeSeekBar: SeekBar, fontSizeTextView: TextView?)
    {
        fontSizeSeekBar.max = maxSize
        fontSize = customSharedPreference.getInt(key, defaultFontSize)
        fontSizeSeekBar.progress = fontSize
        fontSizeSeekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener
        {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean)
            {
                fontSize = progress
                val text = context.resources.getString(R.string.fontSize) + ": " + fontSize
                fontSizeTextView!!.text = text
            }

            override fun onStartTrackingTouch(seekBar: SeekBar)
            {
                //Do nothing
            }

            override fun onStopTrackingTouch(seekBar: SeekBar)
            {
                saveFontSizePreference(key, fontSize)
            }
        })
    }

    private fun setFontSizeTextView(fontSizeTextView: TextView?)
    {
        val text = context.resources.getString(R.string.fontSize) + ": " + fontSize
        fontSizeTextView!!.text = text
    }

    private fun saveFontSizePreference(key: String, fontSize: Int)
    {
        val fontSizePreference = PreferenceManager.getDefaultSharedPreferences(this@FontDialogPreference.context)
        val editor = fontSizePreference.edit()
        editor.putInt(key, fontSize)
        editor.apply()
    }
}
